run_artemis.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+
+# Params - 2 semver strings of the form 1.11.2.3
+# Returns - 0 if equal version, 1 if 1st param is a greater version number, 2 if 2nd param has greater version number
+checkver () {
+    if [[ $1 == $2 ]]
+    then
+        return 0
+    fi
+    local IFS=.
+    local i ver1=($1) ver2=($2)
+    for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
+    do
+        ver1[i]=0
+    done
+    for ((i=0; i<${#ver1[@]}; i++))
+    do
+        if [[ -z ${ver2[i]} ]]
+        then
+            ver2[i]=0
+        fi
+        if ((10#${ver1[i]} > 10#${ver2[i]}))
+        then
+            return 1
+        fi
+        if ((10#${ver1[i]} < 10#${ver2[i]}))
+        then
+            return 2
+        fi
+    done
+    return 0
+}
+
+STOP=0
+START=1
+DEPLOY=0
+for option in "$@"; do
+    case $option in
+        -b=*|--beamline=*)
+            BEAMLINE="${option#*=}"
+            shift
+            ;;
+        -v=*|--version=*)
+            VERSION="${option#*=}"
+            shift
+            ;;
+        --stop)
+            STOP=1
+            ;;
+        --no-start)
+            START=0
+            ;;
+        --deploy)
+            DEPLOY=1
+            ;;
+        --help|--info)
+            echo "Options"
+            echo "  -b, --beamline=BEAMLINE Overrides the BEAMLINE environment variable with the given beamline"
+            echo "  -v, --version=VERSION   Specifies the artemis version number to deploy. Option should be given in the form 0.0.0.0"
+            echo "                          Will check git tags and use the lastest version as a default if no version is specified."
+            echo "                          Unused outside of deploy operation."
+            echo " "
+            echo "Operations"
+            echo "  --stop                  Used to stop a currently running instance of Artemis. Will override any other operations"
+            echo "                          options"
+            echo "  --deploy                Used to update and install a new version of Artemis."
+            echo "  --no-start              Used to specify that the script should be run without starting the server."
```

Comment:
**nit**: This is a bit weird cos running the script as `run_artemis --no-start` basically does nothing so I guess `--no-start` only really makes sense in conjunction with `--deploy` but I think it's fine.

Reply:
I could also use --deploy-no-start instead of the --no-start flag to specify a deployment without a server start, but neither is particularly nice if we want that option to be available. If you have any other ideas for it I'd be glad to do the implementation.

Reply:
No, I can't really think of any cleaner way necessarily, just highlighting it as odd behavior. We'll leave it as and maybe come up with a better idea for it in the future.